File: 13 Document Manager API/SWDMTableExport/SolidWorksMacro.vb
Imports SolidWorks.Interop.sldworks
Imports SolidWorks.Interop.swdocumentmgr


Partial Class SolidWorksMacro

  Public Sub main()
    Dim MyDM As New MyDocMan

    'open the document for read/write
    Dim dmDoc As SwDMDocument
    Dim FilePath As String
    FilePath = MyDM.BrowseForDoc()
    dmDoc = MyDM.OpenDoc(FilePath, False)
    If dmDoc Is Nothing Then Exit Sub

    Dim tableText As String
    tableText = MyDM.GetBOMTableText(dmDoc)

    'close the document
    dmDoc.CloseDoc()

    'Stop 'look at the tabletext in the watch text visualizer
    Dim bomFile As String = IO.Path.ChangeExtension( _
      dmDoc.FullName, ".BOM.txt")
    My.Computer.FileSystem.WriteAllText(bomFile, tableText, True)

  End Sub

    ''' <summary>
    ''' The SldWorks swApp variable is pre-assigned for you.
    ''' </summary>
    Public swApp As SldWorks


End Class

File: C# Examples/13 Document Manager API/SWDMTableExport/SolidWorksMacro.cs
using System;
using SolidWorks.Interop.sldworks;
using SolidWorks.Interop.swdocumentmgr;
using SWDMProperties;
using System.IO;

namespace SWDMTableExport
{
    public partial class SolidWorksMacro
    {
        public void Main()
        {
            MyDocMan MyDM = new MyDocMan();

            //open the document for read/write
            SwDMDocument dmDoc;
            string filePath = MyDM.BrowseForDoc();
            dmDoc = MyDM.OpenDoc(filePath, false);
            if (dmDoc == null)
            {
                return;
            }

            string tableText = MyDM.GetBOMTableText((SwDMDocument10)dmDoc);

            //close the document
            dmDoc.CloseDoc();

            throw new ArgumentNullException("Look at tableText in the watch Text Visualizer");
            string bomFile = Path.ChangeExtension(
                dmDoc.FullName, ".BOM.txt");
            File.WriteAllText(bomFile, tableText);

            return;
        }

        // The SldWorks swApp variable is pre-assigned for you.
        public SldWorks swApp;

    }
}

